src/pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -99,7 +99,7 @@ def log_metrics(self, metrics: _OUT_DICT, step: Optional[int] = None) -> None:
         if step is None:
             # added metrics for convenience
             scalar_metrics.setdefault("epoch", self.trainer.current_epoch)
-            step = self.trainer.fit_loop.epoch_loop._batches_that_stepped
+            step = self.trainer.global_step - 1
```

Comment:
this isn't BC if multiple optimizers are used.
similar issue: https://github.com/Lightning-AI/lightning/issues/12274

we need to come up with a better solution.